### DISCUSSION TITLE
Albums sorting do not work

### DISCUSSION THREAD

**Discussion by arthuragone** [photoprism/photoprism#3708] [Q&A]
#### 1. What is not working as documented?

When I try to sort albums, I don't work as expected. 

#### 2. How can we reproduce it?

1. Go to 'Albums'
2. Click on 'Newest first' sorting option
3. Scroll down to see every album and their order
4. Click on 'Recentely add' sorting option
5. See error, nothing changed between thoses options

#### 3. What behavior do you expect?

Newest first sorting option should sort albums by inner photo creation date. Album with newer photos first, then album with older photo. 

#### 4. What could be the cause of your problem?

I have no idea

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: PhotoPrism® CE Build 230719-73fa7bbe8

(b) Database Type & Version: MariaDB

(c) Operating System Types & Versions: Linux Ubuntu

(d) Browser Types & Versions: Brave

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?

#### 7. On what kind of device is PhotoPrism installed?

NUC

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

Everything is behind Caprover

**Comment by graciousgrey** [comment]
This works as intended. The "newest first" filter sorts by the year/month/day property of an album. For manually managed albums and state albums, no album year/month/day is defined. In this case "newest first" is the same as "recently added".
Calendar and some moment albums do have a defined year/month, for those albums "newest first" and "recently added" are not the same.

Removing the redundant sort option or adding logic to define year/month/day properties for manually managed albums and state albums is one of the tasks on our to-do list.

**Comment by michaelsapsed** [comment]
I'm glad this issue has been raised as it's been anoying me for a while. I create manual albums for holidays and events and I'd expect to be able to sort albums so those with older photos apear as the older albums and those with newer photos included appear as newer albums. At the moment that's not happending and is ignoring the dates of the photos and sorting albums using the creation or edit date of the album. Looking forward to when this can be resolved.
